internal/archive/archive_test.go

```diff
@@ -302,6 +302,38 @@ func testArchive(t *testing.T, when spec.G, it spec.S) {
 					verify.nextSymLink("/nested/dir/dir-in-archive/sub-dir/link-file", "../some-file.txt")
 				}
 			})
+
+			when.Focus("files are compressed in fat (MSDOS) format", func() {
+				it.Before(func() {
+					src = filepath.Join("testdata", "fat-zip-to-tar.zip")
+				})
+
+				it("writes a tar to the dest dir with 0777", func() {
+					fh, err := os.Create(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+
+					tw := tar.NewWriter(fh)
+
+					err = archive.WriteZipToTar(tw, src, "/nested/dir/dir-in-archive", 1234, 2345, -1, true, nil)
+					h.AssertNil(t, err)
+					h.AssertNil(t, tw.Close())
+					h.AssertNil(t, fh.Close())
+
+					file, err := os.Open(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+					defer file.Close()
+
+					tr := tar.NewReader(file)
+
+					verify := tarVerifier{t, tr, 1234, 2345}
+					// FAT stores names in uppercase: https://superuser.com/a/1297675
+					verify.nextFile("/nested/dir/dir-in-archive/SOME-FIL.TXT", "some-content", 0777)
+					verify.nextDirectory("/nested/dir/dir-in-archive/SUB-DIR", 0777)
+					// FAT doesn't support links: https://unix.stackexchange.com/q/492698
```

Comment:
I think I understand why this comment is here, at least looking at https://github.com/buildpacks/pack/blob/82549aa7c8980ed7a42b0c8947c0a37d6d1edade/internal/archive/archive_test.go#L118. Given that the fixture FAT wasn't created with a symlink (right?), I wonder if it's necessary? Maybe we don't need to have `LINK-FIL` at all?

Reply:
Interesting. I just zipped the dir that is in the testdata, I wanted it to be relatively similar to the `zip-to-tar.zip` file that is in the testdata directory. We can create it without the link-file though

Reply:
I'm on the fence about this. I see how we got here but in reflection would we intentionally test this if we weren't reusing pre-existing test data. I mean it's good to know but probably not something valuable to test. I'm leaning towards removing it as well. Any strong opposition @dfreilich?

Reply:
No, I agree that it's not especially valuable. I'll take it out